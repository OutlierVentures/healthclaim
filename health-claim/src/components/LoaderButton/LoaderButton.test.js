import React from "react";
import renderer from "react-test-renderer";
import LoaderButton from "./LoaderButton";

describe("LoaderButton", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<LoaderButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render with disabled set to true", () => {
    const tree = renderer.create(<LoaderButton disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render with loading text if isLoading", () => {
    const tree = renderer
      .create(<LoaderButton loadingText="LOADING" isLoading />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render with normal text when not loading", () => {
    const tree = renderer.create(<LoaderButton loadingText="LOADING" />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
