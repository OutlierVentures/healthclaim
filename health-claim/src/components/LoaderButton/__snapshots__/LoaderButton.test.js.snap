// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderButton should render correctly 1`] = `
<button
  className="LoaderButton  btn btn-default"
  disabled={false}
  type="button"
/>
`;

exports[`LoaderButton should render with disabled set to true 1`] = `
<button
  className="LoaderButton  btn btn-default"
  disabled={true}
  type="button"
/>
`;

exports[`LoaderButton should render with loading text if isLoading 1`] = `
<button
  className="LoaderButton  btn btn-default"
  disabled={true}
  type="button"
>
  <span
    className="spinning glyphicon glyphicon-refresh"
  />
  LOADING
</button>
`;

exports[`LoaderButton should render with normal text when not loading 1`] = `
<button
  className="LoaderButton  btn btn-default"
  disabled={false}
  type="button"
/>
`;
